### PR TITLE
fix: handle v2025.10 structured color and dimension values in formatters

### LINKED ISSUE #17: converting v2025.10 colors fails
From this:
```
{
	"$schema": "https://www.designtokens.org/schemas/2025.10/format.json",
	"color": {
		"$type": "color",
		"red": {
			"50": {
				"$value": {
					"colorSpace": "oklch",
					"components": [0.988281, 0.0046875, 20],
					"alpha": 1
				}
			}
		}
	}
}
```

To this:
```
/* Generated by asimonim */
/* Do not edit manually */

:root {
  --color-red-50: map[alpha:1 colorSpace:oklch components:[0.988281 0.0046875 20]];
}
```

Using this command: `asimonim convert --format css --output foo.css src/tokens/colors.json`

### PR DESCRIPTION
**Pull request by bennypowers**
## Summary

Fixes #17 — v2025.10 structured color values (objects with `colorSpace`, `components`, `alpha`) were being rendered as Go map literals (`map[alpha:1 colorSpace:oklch ...]`) instead of proper CSS color functions.

- **CSS/SCSS formatters**: Parse structured color maps through `common.ParseColorValue()` to produce proper CSS functions (`oklch()`, `color()`, `hsl()`, etc.) and format structured dimension maps as `{value}{unit}`
- **Android formatter**: Use `tok.DisplayValue()` which already handles structured values correctly
- **Swift formatter**: Add structured dimension handling with type-safe formatting and comment injection protection
- **Schema detection**: Improve duck-typing to detect structured `$value` objects with `colorSpace` even when `$type` is inherited from a parent group; accept `2025.10/format.json` schema URL variant
- **Safety**: Add JSON fallback for any remaining map/slice values to prevent Go literals in output; nil guards on dimension values

## Test plan

- [x] Unit tests for `ToCSSValue` covering all 14 DTCG color spaces (srgb, oklch, oklab, hsl, hwb, lab, lch, display-p3, a98-rgb, prophoto-rgb, rec2020, xyz-d50, xyz-d65, srgb-linear)
- [x] Unit tests for structured dimensions (px, rem, em)
- [x] Fixture test with golden file for v2025.10 CSS output
- [x] SCSS formatter tests for structured colors and dimensions
- [x] Swift formatter tests for structured colors and dimensions
- [x] Android formatter tests for structured colors and dimensions
- [x] Regression tests reproducing exact issue #17 scenario
- [x] Edge case tests: nil values, missing units, comment injection
- [x] `make lint` and `make test` pass (all existing tests unaffected)
- [x] End-to-end verification: `asimonim convert --format css` and `asimonim list` produce correct output

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Expanded Android color output to produce platform-compatible hex (including alpha) and broader color-space support; improved Swift/CSS/SCSS handling for structured colors and dimensions.
  * Added a generic JSON fallback to avoid raw map/slice literals in formatted output.

* **Tests**
  * Many new and expanded fixtures/tests covering color spaces, alpha handling, dimensions, schema variants, and regressions.

* **Chore / Refactor**
  * Schema V2025.10 support, detection tweaks, centralized testdata utilities, and extensive test data added.

* **Documentation**
  * Testing guidance revised toward fixture-driven workflows.

### CODE REVIEW

**Comment by bennypowers** [comment]
@coderabbitai review

**Comment by bennypowers** [comment]
@coderabbitai resume

**Comment by bennypowers** [comment]
@coderabbitai review

File: testutil/fixtures.go [hunk 1]
```diff
@@ -81,47 +71,100 @@ func NewFixtureFS(t *testing.T, fixtureDir string, rootPath string) *mapfs.MapFi
...
+	for i := range maxParentProbe {
+		prefix := strings.Repeat("../", i)
+		candidate := filepath.Join(prefix+"testdata", relPath)
+		if _, err := os.Stat(candidate); err == nil {
+			return candidate
 		}
 	}
-	t.Fatalf("Failed to read fixture %s (tried all paths)", fixturePath)
-	return nil
+	return ""
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply testutil/fixtures.go]: Not fixing. This is test infrastructure, not a security boundary. All callers pass hardcoded string literals in test files. Adding path validation here would be defensive against a scenario that cannot happen in practice (test code injecting malicious paths into its own fixture loader).
